### DISCUSSION TITLE
Plex URL is invalid

### DISCUSSION THREAD

**Discussion by Eddi101** [meisnate12/Plex-Meta-Manager#490] [Q&A]
![2021-12-05 22 42](https://user-images.githubusercontent.com/86011893/144766819-98e4de11-283c-4444-a98e-5b6b7b646a7d.png)

{from config.yml}

plex:   
  url: https://10.10.10.10:32400
  token: reVb#############
  clean_bundles: false
  empty_trash: false
  optimize: false
  timeout: 60

**Comment by muff7** [comment]
Had this issue myself on my Unraid setup. 

I changed the url to localhost:32400 and the script runs now.

**Comment by meisnate12** [comment]
i dont know your setup but your pmm cannot ping that url to reach plex you can try muffs suggestion but it has to do with your network configuration

**Comment by gitttt54** [comment]
Can we elaborate ?

_What_ network configuration might be the problem?

Unraid itself, or the Docker container's or the Plex remote settings? or all?

I have tried the following:

updating docker containter: 

**Network Type**: `Host` or `Bridge` (default was bridge as seen in [here](https://camo.githubusercontent.com/efbd7e3f3387f7c445fe1cebd1f4114299e38d061fed851abb69bc03652114c0/68747470733a2f2f692e696d6775722e636f6d2f327938525054742e706e67)).

**Privileged**: `on`/`off` (off by default)

and also in **config.yml**
`url: https://192.168.178.103:32400` (my unraid host ip, up and running. With plex docker from plex-inc in https mode)
or
`url: https://localhost:32400` (did not work)
or just
`url: localhost:32400` (gave another error: _| Plex Error: No connection adapters were found for 'localhost:32400/' |_)

All above combinations throw the same error: _| Plex Error: Plex url is invalid |_

Unfortunately, I can not ping inside the docker console, to check if the host is reachable. 
There seems to be no ping installed. Dunno, if this is an unraid or docker issue.

I have no idea atm, what else I can try or which logs I should read. Nor, what else I can hand over to you guys.

I run the docker container via `-r` in **Post Arguments**
[config.yml](https://pastebin.com/jWG8g0wb)
[meta.log](https://pastebin.com/JUZ2vgQ6)

**Comment by alexricher** [comment]
Hi gitttt54,

I also had the same issue on Unraid installing this docker. My plex is not on the same IP of my Unraid host, so the "use localhost:32400" trick wouldn't work for me either.

I tried pinging my Plex server from the console of the docker directly, and it wouldn't work. Solution for me was to set my network type as "br0" so it has its own IP address assigned to it. I could then access the Plex url successfully. 

Hopefully, that solves your issue as well.

Good luck and have a great day!

Alex

**Comment by gitttt54** [comment]
Thanks, 
questions:

I guess "br0" is bridge? I only have eth0, eth2 shown as nics.
See here:
<img width="239" alt="Bildschirmfoto 2021-12-27 um 16 13 37" src="https://user-images.githubusercontent.com/96403800/147484591-99e4c08e-6913-493b-a20f-e48e3c155610.png">


So on **what** docker container did you switch your network type to bridge mode? plex-docker or plex-meta-manager? or both?

Thank you very much for your help!

**Reply by alexricher** [reply]
Hi,

Yeah, br0 is my network card in my case, but from the look of your drop-down list, I'm guessing you might have 2 network cards in your system. In this case, take the one that is connected to your LAN (local network, i.e. most likely eth0 or eth2). You don't need to provide a specific ip underneath unless you want a static ip.

Try that and see if this works. 👍😊

Alex

Edit : forgot to answer your other question. My plex is also in "custom: br0" with a static ip (ex: 192.168.1.199). I bet I wouldn't have this problem if my plex setup was in bridge mode and we could use the localhost solution provided above in the thread.

**Comment by gitttt54** [comment]
Ok, thanks for clarifying!

unfortunately, no change, when I use the above suggested settings now.

My Unraid IP is `192.168.178.103` via https
My Plex Docker IP is `192.168.178.241` via https
I start the Script from the Plex-Meta-Manager Console: `python plex_meta_manager.py -r `
I also restart the Plex-Meta-Manager Docker container after a change in the config.yml.

My Configs:

config.yml
<img width="427" alt="Bildschirmfoto 2021-12-27 um 19 12 31" src="https://user-images.githubusercontent.com/96403800/147497113-9815cadb-3d01-4b7d-87fd-46c52cc4982e.png">
or another test config.yml
<img width="334" alt="Bildschirmfoto 2021-12-27 um 19 25 43" src="https://user-images.githubusercontent.com/96403800/147497925-7b430a50-7878-4cfc-aee2-b7a4fb8fde73.png">



plex docker:
<img width="1121" alt="Bildschirmfoto 2021-12-27 um 19 08 56" src="https://user-images.githubusercontent.com/96403800/147497117-1b96397d-7286-4f52-bd04-9bbdfd58d3dc.png">

plex-meta-manager docker:
<img width="1109" alt="Bildschirmfoto 2021-12-27 um 19 10 05" src="https://user-images.githubusercontent.com/96403800/147497119-eb8aeac6-811d-45dd-9a30-30f93deff32c.png">

output log:
<img width="560" alt="Bildschirmfoto 2021-12-27 um 19 17 19" src="https://user-images.githubusercontent.com/96403800/147497409-710f415b-f90a-4222-aded-88a6e7d1fa97.png">

Also, my Unraid Network Config:

<img width="1209" alt="Bildschirmfoto 2021-12-27 um 19 28 06" src="https://user-images.githubusercontent.com/96403800/147498117-6970f376-b2a4-47a7-a605-1b7ef1630c9a.png">
<img width="1464" alt="Bildschirmfoto 2021-12-27 um 19 28 25" src="https://user-images.githubusercontent.com/96403800/147498114-4632094e-0c49-4fa4-88c1-fcef37bf2783.png">

I can not see, why the error PLEX URL INVALID is coming.

Unfortunately, I can not test with a ping, because in the plex-meta-manager docker is no ping executable found.

Maybe there is something like a "debug" modus?

**Comment by izeen** [comment]
i have the same issue but i think a key piece of information is missing from the above that can be found through the logs. By default plex does not allow insecure connections (HTTP) and since we are connecting on the local IP and through HTTPS there will be a certificate error which can be spotted in the logs:

`File "/usr/local/lib/python3.9/ssl.py", line 420, in match_hostname
                                                                 |     raise CertificateError("hostname %r "
                                                                 | ssl.SSLCertVerificationError: ("hostname '192.168.XXX.XXX' doesn't match '*.XXXXXXXXXXXXXXXXXXX.plex.direct'",)`
                                                                 
i've seen from other utilities in the past where you could skip SSL verification by adding it to the configuration (ombi used to have it and way back tautulli but they have both changed their method of auth against plex), is there an equivalent setting here or am i out on deep water and should configure it differently?  Thanks!

**Reply by gitttt54** [reply]
Good point, but I  have no clue...

There is no (at least, I did not find any) documentation in the config.yml file about https/certificates, so I am under the impression, we can easily switch from HTTP to HTTPS by adding just the "S". 

The rest, IMHO, will happen in the script, and does not need any user activity...
IF there are any additional steps for HTTPS, what are they and where do we have to take care? 

I guess, that question can only be answered by the developer...

**Reply by izeen** [reply]
I would much rather make plex-meta-manager a tad less secure by being able to disable SSL hostname verification with a config switch without downgrading the security of plex. 

There are older plex clients out there which do not support encrypted communications that would be able to communicate unencrypted over WAN with the plex setting: secure connection -> preferred. That makes it so i cannot change this setting for anything.

since radarr, sonarr and a lot of other integrations can talk with plex just fine over https i guess the auth process could be change to use the same method as those systems where they use plex auth and some secure cookie (im just guessing here, not a developer unfortunately so i do not know how it works or how hard it would be to implement).


@meisnate12 Any thoughts ?

**Reply by meisnate12** [reply]
Im not very good with SSL stuff tbh but i just pushed a change to the develop branch that may do exactly what youre asking for. Can you test develop and see if youre still getting the SSL error. the develop version should be `1.14.1-1722`

**Reply by izeen** [reply]
Develop works now without problems with the exact conf that didnt work before!

Some thoughts (without checking what changes were done), its probably not wise to have SSL host name verification off permanently since one can run your own custom cert and domain to access plex, and that setup works already since the host name is correct then and here hostname should be verified to avoid possible man in the middle attacks or other nasty things. 

This should only apply to people accessing plex through https only and where plex-meta-manager connects to plex on LAN side of things. So if possible to implement with some kind of command line switch or conf option would be the best so its an option not a default, again without me knowing how things work in the backend nor how the change was implemented or what other security concerns it might raise.

If this is deemed not worthy/not feasible of implementing i would recommend adding to the documentation that If Secure connections is set to required in plex, and one is not connecting over WAN to custom domain + own cert. Plex-meta-manager wont be able to connect. At least then people will know instantly that there is something in the setup preventing them from connecting and maybe save them from a bit of troubleshooting. I think that many people run plex with the setting require secure connections since people are a bit anal when it comes to plex and what data potentially gets leaked to ISP or other parties due to not being encrypted.

Thanks for taking the time to respond and looking into this.

**Reply by meisnate12** [reply]
yea that can be done with ease

btw the change added very little it went from 

```python
        self.session = requests.Session()
```

to

```python
        self.session = requests.Session()
        self.session.verify = False
        if self.session.verify is False:
            import urllib3
            urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```

and the last three lines are just to disable a warning.

**Reply by izeen** [reply]
Great!
Looks good to me!

Thanks again 👍

**Reply by meisnate12** [reply]
ok now youll have to have `verify_ssl: true` under your global settings set that to false to disable the verification

**Reply by gitttt54** [reply]
> ok now youll have to have `verify_ssl: true` under your global settings set that to false to disable the verification

Nice! I will add that line to my config.yml and test it, when a new stable version is updated by the docker image.

**Reply by izeen** [reply]
Just pulled the latest stable docker container and added the verify_ssl: false to the conf, works perfectly! Thanks again!

**Reply by gitttt54** [reply]
> Just pulled the latest stable docker container and added the verify_ssl: false to the conf, works perfectly! Thanks again!

Same here. Now it works! Thank you for the magic!!!

**Comment by rotj** [comment]
I had the same error on Docker for Windows.

Instead of localhost or the ip, I tried host.docker.internal and it worked.

`url: http://host.docker.internal:32400`

**Reply by JoseFilion** [reply]
THIS WORKED THANK YOU!!!

**Reply by Persnickety-Econ** [reply]
yes, thanks for this. worked perfectly and I didn't have to change the "verify_ssl:" value nor did I have to change my Plex/Setting/Network/Secure Connections settings.

**Reply by that1itguy** [reply]
I'm still new to this and learning. When you say url: http://host.docker.internal:32400

Would that mean something like url: http://192.168.3.20.docker.192.168.3.110:32400 or is my syntax wrong?

Would the 'host' be my dockers IP address and internal being plex IP address?

**Reply by chazlarson** [reply]
No, enter exactly that:

```
plex:
  url: http://host.docker.internal:32400/
  token: WHATEVERWHATEVER
```

`host.docker.internal` tells docker to connect to the host machine's IP, whatever it is.

**Reply by that1itguy** [reply]
I am using Docker in Synology and my plex is on another server.
My dockers IP is 192.168.3.20 while Plex is hosted at 192.168.3.110.

I did try using http://192.168.3.110:32400 as the URL but says it's unable to connect as I'm assuming dockers internal IP can't ping 192.168.3.110?

**Reply by chazlarson** [reply]
Oh, then sure, this doesn't apply.

You need to [at a minimum] put the PMM docker on the host network so it can see other machines on the LAN.

**Comment by plexamby** [comment]
I had the same problem and the problem was resolved by changing Plex/Setting/Network/Secure Connections to Preferred

![Screen Shot 2022-04-21 at 8 00 28 PM](https://user-images.githubusercontent.com/104181701/164587804-975272dd-0777-4ef2-b6c9-1952a819eb24.png)

**Reply by SebboGit** [reply]
This is the only thing that works for me. I can ping my plex server within the plex-meta-manager container, I've tried all of the IP addresses mentioned above, I've set `verify_ssl` to false but nothing has work until I've changed Secure connections to "Preferred".

**Reply by NMGMarques** [reply]
Winner! Winner! Chicken dinner! Had to be the very last solution, huh? Thanks. Was going nuts!

**Reply by humdinger44** [reply]
This comment is still solving problems today. Thank you.
